components/language_filter/src/main.py

```diff
@@ -0,0 +1,72 @@
+"""A component that filters text based on the language."""
+import logging
+
+import fasttext
+import pandas as pd
+
+from fondant.component import PandasTransformComponent
+from fondant.logger import configure_logging
+
+configure_logging()
+logger = logging.getLogger(__name__)
+
+
+class LanguageIdentification:
+    """A class for language detection using FastText."""
+
+    def __init__(self, language, model_path: str = "lid.176.ftz"):
+        """
+        Initializes the LanguageDetect class.
+
+        Args:
+           language (str): language to filter on
+           model_path (str): The path to the FastText language identification model.
+        """
+        pretrained_lang_model_weight_path = model_path
+        self.language = language
+        self.model = fasttext.load_model(pretrained_lang_model_weight_path)
+
+    def predict_lang(self, text: str):
+        """
+        Detects the language of a text sequence.
+
+        Args:
+            text (str): The text for language detection.
+
+        Returns:
+            str: The predicted language label.
+        """
+        predictions = self.model.predict(text, k=1)
+        return predictions[0][0]
+
+    def is_language(self, row):
+        """Predict if text of a row is written in the defined language."""
+        return self.language in self.predict_lang(row["text"])
+
+
+class LanguageFilterComponent(PandasTransformComponent):
+    """Component that filter columns based on provided language."""
+
+    def setup(self, *, language):
+        """Setup language filter component."""
+        self.lang_detector = LanguageIdentification(language)
+
+
+    def transform(self, dataframe: pd.DataFrame) -> pd.DataFrame:
+        """
+        Args:
+            dataframe: Pandas dataframe.
+            language: Only keep text passages which are in the provided language.
+
+        Returns:
+            Pandas dataframe
+        """
+        mask = dataframe.apply(
+            lambda row: self.lang_detector.is_language(row), axis=1)
```

Comment:
```suggestion
        mask = dataframe.apply(self.lang_detector.is_language, axis=1)
```
I think you can drop the lambda.